plugins/sync.js

```diff
@@ -83,73 +84,93 @@ window.plugin.sync.RegisteredMap = function(options) {
 window.plugin.sync.RegisteredMap.prototype.updateMap = function(keyArray) {
   var _this = this;
   try {
-    this.lastUpdateUUID = this.uuid;
-
-    $.each(keyArray, function(ind, key) {
-      var value = window.plugin[_this.pluginName][_this.fieldName][key];
-      if(typeof(value) !== 'undefined') {
-        _this.map[key] = value;
-      } else {
-        delete _this.map[key];
+    this.lastUpdateUUID = this.uuid;                        // use this clients uuid
+    $.each(keyArray, function(ind, key) {                   // find all changed elements from the array
+      var value = window.plugin[_this.pluginName][_this.fieldName][key]; 
+      if(typeof(value) !== 'undefined') {                  
+        _this.map[key] = value;                             // replace element in the array
+      } else {                                              // or
+        delete _this.map[key];                              // remove it from the array
       }
     });
   } finally {
-    _this.dataStorage.saveFile(_this.prepareFileData());
+    _this.dataStorage.saveFile(_this.prepareFileData());    // save data and UUID to drive
   }
 };
 
-window.plugin.sync.RegisteredMap.prototype.isUpdatedByOthers = function() {
-  var remoteUUID = this.lastUpdateUUID.toString();
-  return (remoteUUID !== '') && (remoteUUID !== this.uuid);
-};
-
+// concatenate a the filename for the drive "pluginName[fieldName]"
 window.plugin.sync.RegisteredMap.prototype.getFileName = function() {
   return this.pluginName + '[' + this.fieldName + ']'
 };
 
+// initialize access to the sync-file on drive
+//   callback (optional) - to be performed on success
 window.plugin.sync.RegisteredMap.prototype.initFile = function(callback) {
   var assignIdCallback, failedCallback, _this;
-  _this = this;
+  _this = this;                                             // have _this point to this Object
 
-  assignIdCallback = function(id) {
+  // file access granted, perform callback-function
+  assignIdCallback = function(id) {                         // where does this id come from?
     _this.forceFileSearch = false;
-    if(callback) callback();
+    if(callback) callback();                                // do the callback passed to .initfile
   };
 
+  // file access not possible
   failedCallback = function(resp) {
     _this.initializing = false;
     _this.failed = true;
     plugin.sync.logger.log(_this.getFileName(), 'Could not create file. If this problem persist, delete this file in IITC-SYNC-DATA-V3 in your Google Drive and try again.');
   };
 
-  this.dataStorage = new plugin.sync.DataManager({'fileName': this.getFileName(),
-                                                    'description': 'IITC plugin data for ' + this.getFileName()});
-  this.dataStorage.initialize(this.forceFileSearch, assignIdCallback, failedCallback);
+  this.dataStorage = new plugin.sync.DataManager(           // create a new dataStorage with
```

Comment:
```suggestion
  this.dataStorage = new plugin.sync.DataManager(
```
The interesting part is "what is a DataManager" - and this will hopefully answered later.